crypto/s2n_dhe.c

```diff
@@ -102,6 +102,7 @@ static int s2n_check_pub_key_dh_params(struct s2n_dh_params *dh_params)
 static int s2n_set_p_g_Ys_dh_params(struct s2n_dh_params *dh_params, struct s2n_blob *p, struct s2n_blob *g,
                                     struct s2n_blob *Ys)
 {
+    ENSURE_POSIX((p->size <= INT_MAX) && (g->size <= INT_MAX) && (Ys->size <= INT_MAX), S2N_ERR_INTEGER_OVERFLOW);
```

Comment:
@camshaft should I break this down into 3 `ENSURE_POSIX`?

Reply:
Good catch. Definitely a lot clearer!

Reply:
Great, implemented as suggested.